### PR TITLE
Add HumanHandle validation to rust implementation

### LINKED ISSUE #2666: Missing validation for rust implementation of `HumanHandle`
This could be used as a substitute to `email.utils.parseaddr` from python: https://docs.rs/email-format/0.8.0/email_format/rfc5322/types/struct.NameAddr.html

### PR DESCRIPTION
**Pull request by vxgmichel**
Fix #2666

### CODE REVIEW

**Review by touilleMan — APPROVED** [review]
👍

**Review by TimeEngineer — APPROVED** [review]

File: oxidation/libparsec/crates/types/src/id.rs [hunk 1]
```diff
@@ -188,24 +189,28 @@ impl std::fmt::Display for HumanHandle {
...
         let email = email.nfc().collect::<String>();
         let label = label.nfc().collect::<String>();
 
-        // TODO: how to check the email  easily ?
-        if email.is_empty() || email.len() >= 255 {
+        if !EmailAddress::is_valid(&email, None) || email.len() >= 255 {
             return Err("Invalid email address");
         }
-        if label.is_empty() || label.len() >= 255 {
+        // According to https://www.rfc-editor.org/rfc/rfc5322#section-3.2.3, these special characters are not allowed
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/types/src/id.rs:201]: not allowed for what ?

the following comment indicate that the label is only informative, so why the restriction ?

```rust
// Label is purely informative
pub label: String,
```

Comment (TimeEngineer) [reply oxidation/libparsec/crates/types/src/id.rs:201]: The label is equivalent to the display name when sending an email
And in test `[Test]` as label should lead to error

Comment (FirelightFlagboy) [reply oxidation/libparsec/crates/types/src/id.rs:201]: So maybe the comment is misleading here (the one for the label field)

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**